api/src/main/java/org/openmrs/util/OpenmrsUtil.java

```diff
@@ -1101,8 +1101,7 @@ public static String getApplicationDataDirectory() {
 					filepath = OpenmrsConstants.APPLICATION_DATA_DIRECTORY_FALLBACK_UNIX + File.separator + openmrsDir;
 				}
 			} else {
-				filepath = System.getProperty("user.home") + File.separator + "Application Data" + File.separator
-				        + "OpenMRS";
+				filepath = System.getenv("appdata") + File.separator + "OpenMRS";
 				if (!canWrite(new File(filepath))) {
```

Comment:
Probably this condition is never caught in tests and hence the failing coverage check.

Reply:
@theanandankit First of all, thank you for this! It's important that we get this right. I would be concerned that we might be breaking existing Windows-based installs without having a proper fallback. Could we implement this as something like:

```java
filepath = System.getProperty("user.home") + File.separator + "Application Data" + File.separator + "OpenMRS";

if (!new File(filepath).exists()) { 
    filepath = System.getenv("appdata") + File.separator + "OpenMRS";
}
```